sdcm/provision/scylla_yaml/auxiliaries.py

```diff
@@ -0,0 +1,155 @@
+LOGGER = logging.getLogger(__file__)
+
+
+@attr.attrs(kw_only=True, order=False, auto_attribs=True, init=True, eq=True, repr=True)
+class SeedProvider:  # pylint: disable=too-few-public-methods
+    class_name: Literal[
+        'org.apache.cassandra.locator.SimpleSeedProvider',
+        'org.apache.cassandra.locator.GossipingPropertyFileSnitch',
+        'org.apache.cassandra.locator.Ec2Snitch',
+        'org.apache.cassandra.locator.Ec2MultiRegionSnitch',
+        'org.apache.cassandra.locator.GoogleCloudSnitch',
+        'org.apache.cassandra.locator.RackInferringSnitch',
+    ]
+    parameters: List[dict] = None
+
+
+@attr.attrs(kw_only=True, order=False, auto_attribs=True, init=True, eq=True, repr=True)
+class ServerEncryptionOptions:  # pylint: disable=too-few-public-methods
+    internode_encryption: Literal['all', 'none', 'dc', 'rack'] = 'none'
+    certificate: str = 'conf/scylla.crt'
+    keyfile: str = 'conf/scylla.key'
+    truststore: str = None
+    priority_string: str = None
+    require_client_auth: bool = False
+
+
+@attr.attrs(kw_only=True, order=False, auto_attribs=True, init=True, eq=True, repr=True)
+class ClientEncryptionOptions:  # pylint: disable=too-few-public-methods
+    enabled: bool = False
+    certificate: str = 'conf/scylla.crt'
+    keyfile: str = 'conf/scylla.key'
+    truststore: str = None
+    priority_string: str = None
+    require_client_auth: bool = False
+
+
+@attr.attrs(kw_only=True, order=False, auto_attribs=True, init=True, eq=True, repr=True)
+class RequestSchedulerOptions:  # pylint: disable=too-few-public-methods
+    throttle_limit: int = None
+    default_weight: int = 1
+    weights: int = 1
+
+
+EndPointSnitchType = Literal[
+    'org.apache.cassandra.locator.SimpleSnitch',
+    'org.apache.cassandra.locator.GossipingPropertyFileSnitch',
+    'org.apache.cassandra.locator.PropertyFileSnitch',
+    'org.apache.cassandra.locator.Ec2Snitch',
+    'org.apache.cassandra.locator.Ec2MultiRegionSnitch',
+    'org.apache.cassandra.locator.RackInferringSnitch'
+]
+
+
+@attr.attrs(kw_only=True, order=False, str=False, init=True)
+class ScyllaYamlBuilderBase(ParamsBuilder):
+    params: Union[SCTConfiguration, dict] = attr.attrib(metadata={'input_param': True}, eq=False)
+
+    @property
+    def _cluster_backend(self) -> str:
+        return self.params.get('cluster_backend')
+
+    @property
+    def _cloud_provider(self) -> Literal['aws', 'gce', 'azure', None]:
+        for provider in ['aws', 'gce', 'azure']:
+            if provider in self._cluster_backend:
+                return provider
+        if self._cluster_backend == "k8s-eks":
+            return 'aws'
+        if self._cluster_backend == "k8s-gke":
+            return 'gce'
+        return None
+
+    @builder_property
+    def _region(self) -> str:
+        if self._cloud_provider == 'aws':
+            self._require_param('region_name', 'Cloud provider is AWS, but not params["region_name"] is provided')
```

Comment:
```but %foo% is NOT provided``` is correct way to say what must be said here. The same below.

Reply:
but no something is provided is also correct way, fixed "not" to "no"